DT-6175 part 4

vesameskanen opened the pull request:
- Refactor Itinerary page jsx rendering by using shared code blocks, don't repeat same  code again and again
- Remove duplicate query
- Shared wrapper for detail view
- Correct spinners while loading
- Correct notifiers and error messages
- Refactor planParamUtil and settings management
- Simplify
- Remove annoying ticket name comments from code

vesameskanen committed:
fix: remove duplicate query

vesameskanen committed:
fix: badly buggy itinerary comparison utility

vesameskanen committed:
chore: used changed queries and utils

vesameskanen committed:
chore: refactor mqtt topic extraction

vesameskanen committed:
chore: remove duplicate div wrapper, no need to use dummy plan

vesameskanen committed:
chore: hook back to lower case

vesameskanen committed:
fix: refactor ItinerarPage DOM rendering thoroughly

- remove if-else maze which exceeds human comprehension
- put shered jsx into variables
- move detail view tab wrapping to new component ItineraryTabs
- remove redundant MobileItineraryWrapper

vesameskanen committed:
Merge branch 'DT-6175-3' into DT-6175-4

vesameskanen committed:
fix: driving distance in P&R button

vesameskanen committed:
Merge branch 'v3' into DT-6175-4

vesameskanen committed:
chore: export named functions

vesameskanen committed:
chore: simplify ItineraryPage

vesameskanen committed:
chore: remove ticket names from code

We want to keep code clean and readable, and use git for documenting
change history.

vesameskanen committed:
chore: simplify Itinerary page more

vesameskanen committed:
fix: don't duplicate function code 3 times, use parameters instead

vesameskanen committed:
chore: refactor ItinerarySummary and WalkDistance components

vesameskanen committed:
chore: simplify and rename

vesameskanen committed:
chore: clean up some components

vesameskanen committed:
fix: spinners

vesameskanen committed:
chore: final touch to itinerary page dom rendering

All building blocks are now shared between mobile and desktop.
Desktop view composition is defined in 24 code lines (was 243 lines).

vesameskanen committed:
chore: simplify declaration of some itinerary page sub components

vesameskanen committed:
fix: move parkride title to same place where other titles are rendered

vesameskanen committed:
fix: don't kick user out from detail view easily

Also, tune spinning for alt itinerary view

vesameskanen committed:
fix: park & ride obeys transport mode settings

vesameskanen committed:
fix: parkride does not always use transit

vesameskanen committed:
fix: do not show alt bar in p&r and bikepublic summary views

vesameskanen committed:
test: jyväskylä tickets

vesameskanen committed:
chore: clean up exporting of some components

vesameskanen committed:
fix: refactor error messages and notifiers

- remove strange duplicate and incorrect notifier
- don't pass existing context fields as props
- cleanup components

vesameskanen committed:
chore: tune itineray page

- add basic loading state because root props.loading sets lazily in refetch
- define driving and biking props correctly for notifiers

vesameskanen committed:
fix: update unit test

vesameskanen committed:
Merge branch 'v3' into DT-6175-4

optionsome commented on app/component/ItineraryPage.js:
```diff
@@ -259,8 +256,6 @@ class ItineraryPage extends React.Component {
           result.bikeAndPublicPlan?.itineraries,
         );
 
-        // show 6 bike + transit itineraries, preferably 3 of both kind.
-        // If there is not enough of a kind, take more from the other kind
```
Why is this removed now?

vesameskanen replied:
Strange. Must be some git merge thing. Restored now.

optionsome commented on app/component/ItineraryPage.js:
```diff
@@ -472,12 +465,10 @@ class ItineraryPage extends React.Component {
-      loadingMore: reversed ? spinnerPosition.bottom : spinnerPosition.top,
-    });
-    this.showScreenreaderLoadingAlert();
+    this.setState({ loadingMore: reversed ? 'bottom' : 'top' });
```
Why do we go back to hardcoding strings here?

vesameskanen replied:
Same as above. A bit worrying. Restored.

optionsome commented on app/component/ItineraryPage.js:
```diff
@@ -623,16 +614,13 @@ class ItineraryPage extends React.Component {
-          viaPoints,
-        ) || // detail view has no itineraries to show
-        (!state.loadingAlt && !this.mapHashToPlan(hash)?.itineraries?.length)
+        // loadingAlt is first undefined, then true and finally false
```
Maybe it's better to define some three state enum instead of having this fuzzy logic.

vesameskanen replied:
Agreed and done.

optionsome commented on app/component/ItineraryPage.js:
```diff
@@ -1149,305 +1082,158 @@ class ItineraryPage extends React.Component {
+        />
+      );
+      const bckBtnFallback =
+        detailView || altTransitHash.includes(hash) ? 'pop' : undefined;
```
this seems a bit weird

vesameskanen replied:
added a comment

optionsome commented on app/component/ItineraryPageUtils.js:
```diff
@@ -358,6 +360,8 @@ export function checkDayNight(iconId, timem, lat, lon) {
   return iconId;
 }
 
+const streetLegModes = ['WALK', 'BICYCLE', 'CAR', 'SCOOTER'];
```
Spelling of these static helper variables could be STREET_LEG_MODES but not that important.

vesameskanen replied:
done

vesameskanen committed:
fix: use enums for alt loading state, add back accidentally lost code changes

vesameskanen committed:
fix: dont't fetch biking if accessibility option is set

vesameskanen committed:
fix: changeHash prop

vesameskanen committed:
fix: refactor planParamsUtil

No need to use a function returning function returning function,
as amusing it is to maximize complexity. Improve names and remove
duplicate parameter.

vesameskanen committed:
fix: refactor settings management

vesameskanen committed:
Merge branch 'v3' into DT-6175-4

vesameskanen committed:
fix: some parametrization fixes and cleanup

vesameskanen committed:
fix: planParamUtil unit tests

vesameskanen committed:
chore: clean up modeutils

vesameskanen committed:
fix: modes for plan query

vesameskanen committed:
chore: tune aria messages

vesameskanen committed:
fix: make streetmode buttons accessible
